modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/Scala3StepFilter.scala

```diff
@@ -12,18 +12,47 @@ import scala.util.Try
 import java.nio.file.Path
 import scala.util.Success
 import scala.util.Failure
+import java.util.Optional
 import ch.epfl.scala.debugadapter.ScalaVersion
 
 class Scala3StepFilter(
     scalaVersion: ScalaVersion,
     bridge: Any,
-    skipMethod: Method
+    skipMethod: Method,
+    formatMethod: Method
 ) extends ScalaStepFilter(scalaVersion) {
-  override protected def skipScalaMethod(method: jdi.Method): Boolean =
+  override protected def skipScalaMethod(method: com.sun.jdi.Method): Boolean = {
     try skipMethod.invoke(bridge, method).asInstanceOf[Boolean]
     catch {
       case e: InvocationTargetException => throw e.getCause
     }
+  }
+
+  private def isAnonFunction(method: jdi.Method): Boolean =
+    method.name.matches(".+\\$anonfun\\$\\d+")
+
+  private def formatAnonFunction(method: jdi.Method): Optional[String] = {
+    val regex = "\\$\\$anonfun.*"
+    var result = method.toString.replaceAll(regex, "").replaceAll("\\$", ".")
+    result = result ++ ".anonfun" ++ "("
+
+    method.argumentTypeNames().forEach(t => result = result ++ t)
+    Optional.of(result ++ ")")
+  }
```

Comment:
All of this is already defined in `ScalaStepFilter` (which is the super class of `Scala3StepFilter`).  It should be used directly there instead.